File: src/leetcode/2025/11/Solution168.java
public class Solution168 {
    public String convertToTitle(int columnNumber) {
        StringBuilder title = new StringBuilder();
        while (columnNumber > 0) {
            columnNumber--; // Adjust for 0-indexing
            int remainder = columnNumber % 26;
            title.append((char) (remainder + 'A'));
            columnNumber /= 26;
        }
        return title.reverse().toString();
    }

    public static void main(String[] args) {
        Solution168 solution = new Solution168();

        // Example 1
        int columnNumber1 = 1;
        String result1 = solution.convertToTitle(columnNumber1);
        System.out.println("Example 1: " + result1);
        // Expected: "A"

        // Example 2
        int columnNumber2 = 28;
        String result2 = solution.convertToTitle(columnNumber2);
        System.out.println("Example 2: " + result2);
        // Expected: "AB"

        // Example 3
        int columnNumber3 = 701;
        String result3 = solution.convertToTitle(columnNumber3);
        System.out.println("Example 3: " + result3);
        // Expected: "ZY"
    }
}

File: src/leetcode/2025/11/Solution168.cpp
#include <string>
using namespace std;

class Solution168
{
public:
    string convertToTitle(int columnNumber)
    {
        string result;
        while (columnNumber > 0)
        {
            columnNumber--; // Adjust for 0-based index
            char ch = 'A' + (columnNumber % 26);
            result = ch + result;
            columnNumber /= 26;
        }
        return result;
    }
};

int main()
{
    // Example 1
    Solution168 solution;
    int columnNumber1 = 1;
    string result1 = solution.convertToTitle(columnNumber1);
    // Expected Output: "A"
    printf("Column Number: %d -> Title: %s\n", columnNumber1, result1.c_str());

    // Example 2
    int columnNumber2 = 28;
    string result2 = solution.convertToTitle(columnNumber2);
    // Expected Output: "AB"
    printf("Column Number: %d -> Title: %s\n", columnNumber2, result2.c_str());

    // Example 3
    int columnNumber3 = 701;
    string result3 = solution.convertToTitle(columnNumber3);
    // Expected Output: "ZY"
    printf("Column Number: %d -> Title: %s\n", columnNumber3, result3.c_str());

    return 0;
}